### PR TITLE
Pkgs data provider schema compliance

### PR DESCRIPTION
**Pull request by MiguelazoDS**
|Related issue|
|---|
|#18003|

## Description

This PR adds the required changes to the different package data providers to be compliant with the defined JSON schema.

- Compilation without warnings in every supported platform
  - [x] Linux
  - [x] Windows
  - [x] MAC OS X
- [x] Source installation
- [x] Review logs syntax and correct language

### CODE REVIEW

File: src/data_provider/qa/test_packages.py
Comment (pereyra-m) [inline src/data_provider/qa/test_packages.py:1]: If the test fails, we can't see the details of the output.
This doesn't happen here for example
https://github.com/wazuh/wazuh/blob/05fd7ab55ebce86f22d9b67c2937d1d4e0fb3b6c/src/data_provider/qa/test_ports.py

File: src/data_provider/src/packages/packageLinuxParserHelper.h [hunk 1]
```diff
@@ -45,11 +46,11 @@ namespace PackageLinuxHelper
         {
             ret["name"] = info.at("Package");
 
-            std::string priority;
-            std::string groups;
-            std::string multiarch;
+            std::string priority {UNKNOWN_VALUE};
```
Comment (pereyra-m) [inline src/data_provider/src/packages/packageLinuxParserHelper.h:49]: Why would we have a default value just for some fields?
In `brewWrapper.h` all the fields have one.

File: src/data_provider/src/packages/pkgWrapper.h [hunk 2]
```diff
@@ -28,7 +28,9 @@ class PKGWrapper final : public IPackageWrapper
 {
     public:
         explicit PKGWrapper(const PackageContext& ctx)
-            : m_architecture{UNKNOWN_VALUE}
+            : m_version{UNKNOWN_VALUE}
```
Comment (pereyra-m) [inline src/data_provider/src/packages/pkgWrapper.h:31]: Why would we have a default value just for some fields?
In `brewWrapper.h` all the fields have one.

File: src/data_provider/src/sysInfoWin.cpp [hunk 3]
```diff
@@ -379,11 +379,17 @@ static void getPackagesFromReg(const HKEY key, const std::string& subKey, std::f
...
+                    packageJson["description"]  = UNKNOWN_VALUE;
+                    packageJson["version"]      = version.empty() ? UNKNOWN_VALUE : std::move(version);
+                    packageJson["groups"]       = UNKNOWN_VALUE;
+                    packageJson["priority"]       = UNKNOWN_VALUE;
+                    packageJson["size"]           = 0;
+                    packageJson["vendor"]       = vendor.empty() ? UNKNOWN_VALUE : std::move(vendor);
+                    packageJson["source"]       = UNKNOWN_VALUE;
+                    packageJson["install_time"] = install_time.empty() ? UNKNOWN_VALUE : std::move(install_time);
+                    packageJson["location"]     = location.empty() ? UNKNOWN_VALUE : std::move(location);
+                    packageJson["architecture"] = architecture.empty() ? UNKNOWN_VALUE : std::move(architecture);
```
Comment (pereyra-m) [inline src/data_provider/src/sysInfoWin.cpp:391]: Considering the lines above, the field `architecture` can't be empty.

**Review by pereyra-m — COMMENTED** [review]
Gj!

**Review by havidarou — APPROVED** [review]

File: src/data_provider/qa/test_packages.py
Comment (pereyra-m) [reply src/data_provider/qa/test_packages.py:1]: The exception already shows the details as you show.
Thank you.

Comment (MiguelazoDS) [reply src/data_provider/qa/test_packages.py:1]: Here are some examples of the output when the tests fail. 

image
image
image

Basically if the field that is causing problem is not null, I'll parse the output of "e" to get the name and format of the package.

File: src/data_provider/src/sysInfoWin.cpp (hunk 3 above)
Comment (MiguelazoDS) [reply src/data_provider/src/sysInfoWin.cpp]: Yeah, the check is useless in that case.

File: src/data_provider/src/packages/pkgWrapper.h (hunk 2 above)
Comment (MiguelazoDS) [reply src/data_provider/src/packages/pkgWrapper.h:31]: Added for vendor and description.

File: src/data_provider/src/packages/packageLinuxParserHelper.h (hunk 1 above)
Comment (MiguelazoDS) [reply src/data_provider/src/packages/packageLinuxParserHelper.h:49]: I narrowed the changes to the ones that fail, you're right.

File: src/data_provider/src/packages/pkgWrapper.h [hunk 4]
```diff
@@ -143,19 +145,23 @@ class PKGWrapper final : public IPackageWrapper
...
                         }
                         else if (line == "<key>CFBundleIdentifier</key>" &&
                                  std::getline(data, line))
                         {
-                            m_description = getValueFnc(line);
+                            auto description = getValueFnc(line);
+                            m_description = description.empty() ? UNKNOWN_VALUE : description;
 
                             std::string vendor;
+                            m_vendor = UNKNOWN_VALUE;
```
Comment (matias-braida) [inline src/data_provider/src/packages/pkgWrapper.h:164]: "m_vendor" already has UNKNOWN_VALUE as default value. It does not need to reassign the same value.

**Review by matias-braida — COMMENTED** [review]
GJ!

File: src/data_provider/src/packages/pkgWrapper.h [hunk 5]
```diff
@@ -143,19 +145,23 @@ class PKGWrapper final : public IPackageWrapper
                         else if (line == "<key>CFBundleShortVersionString</key>" &&
                                  std::getline(data, line))
                         {
-                            m_version = getValueFnc(line);
+                            auto version = getValueFnc(line);
+                            m_version = version.empty() ? UNKNOWN_VALUE : version;
```
[the remark this answers is not in the document]
Comment (MiguelazoDS) [reply src/data_provider/src/packages/pkgWrapper.h]: Yes, It's redundant the assignation of UNKNOWN_VALUE

**Review by pereyra-m — APPROVED** [review]
LGTM!

**Review by matias-braida — APPROVED** [review]
LGTM

**Review by GabrielEValenzuela — APPROVED** [review]
LGTM!

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**